libs/brain/switch/src/lib/brn-switch.component.ts

```diff
@@ -36,48 +35,46 @@ export const BRN_SWITCH_VALUE_ACCESSOR = {
 
 const CONTAINER_POST_FIX = '-switch';
 
+let uniqueIdCounter = 0;
+
 @Component({
 	selector: 'brn-switch',
-	imports: [NgStyle],
+	standalone: true,
 	template: `
-		<input
-			#checkBox
-			tabindex="-1"
-			type="checkbox"
+		<button
+			#switch
 			role="switch"
+			type="button"
+			[class]="class()"
 			[id]="forChild(state().id) ?? ''"
 			[name]="forChild(state().name) ?? ''"
 			[value]="checked() ? 'on' : 'off'"
-			[ngStyle]="{
-				position: 'absolute',
-				width: '1px',
-				height: '1px',
-				padding: '0',
-				margin: -'1px',
-				overflow: 'hidden',
-				clip: 'rect(0, 0, 0, 0)',
-				whiteSpace: 'nowrap',
-				borderWidth: '0',
-			}"
-			[checked]="checked()"
-			[attr.aria-label]="ariaLabel()"
-			[attr.aria-labelledby]="ariaLabelledby()"
-			[attr.aria-describedby]="ariaDescribedby()"
-			[attr.aria-required]="required() || null"
-		/>
-		<ng-content select="brn-switch-thumb" />
+			[attr.aria-checked]="checked()"
+			[attr.aria-label]="ariaLabel() || null"
+			[attr.aria-labelledby]="mutableAriaLabelledby()() || null"
+			[attr.aria-describedby]="ariaDescribedby() || null"
+			[attr.data-state]="checked() ? 'checked' : 'unchecked'"
+			[attr.data-focus-visible]="focusVisible()"
+			[attr.data-focus]="focused()"
+			[attr.data-disabled]="state().disabled()"
+			[disabled]="disabled()"
```

Comment:
I think this should also use the disabled state signal.

```suggestion
			[disabled]="state().disabled()"
```